Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteShaderModuleProxy.cpp

```diff
@@ -48,13 +48,14 @@ RemoteShaderModuleProxy::~RemoteShaderModuleProxy()
 
 void RemoteShaderModuleProxy::compilationInfo(CompletionHandler<void(Ref<PAL::WebGPU::CompilationInfo>&&)>&& callback)
 {
-    auto sendResult = sendSync(Messages::RemoteShaderModule::CompilationInfo());
-    auto [messages] = sendResult.takeReplyOr(Vector<CompilationMessage> { });
-
-    auto backingMessages = messages.map([](CompilationMessage compilationMessage) {
-        return PAL::WebGPU::CompilationMessage::create(WTFMove(compilationMessage.message), compilationMessage.type, compilationMessage.lineNum, compilationMessage.linePos, compilationMessage.offset, compilationMessage.length);
+    auto sendResult = sendWithAsyncReply(Messages::RemoteShaderModule::CompilationInfo(), [callback = WTFMove(callback)](auto messages) mutable {
+        auto backingMessages = messages.map([](CompilationMessage compilationMessage) {
+            return PAL::WebGPU::CompilationMessage::create(WTFMove(compilationMessage.message), compilationMessage.type, compilationMessage.lineNum, compilationMessage.linePos, compilationMessage.offset, compilationMessage.length);
```

Comment:
Is it possible construct `WebGPU::CompilationMessage` directly? Feels strange to me to unpack a struct to pass to a construct to be packed into a struct.

Reply:
Unless you've internalised the C++ Koan of the Big Barking Dog. https://twitter.com/tvaneerd/status/977257752820113413

Reply:
We could pass it directly, but I think the point is to avoid using ::WebGPU::Type inside the PAL layer which is why the unpacking to non-::WebGPU::type is here. As to why there are both the PAL::WebGPU and ::WebGPU seemingly identical types instead of directly using PAL:: types here, I would defer to Myles